pwndbg/commands/__init__.py

```diff
@@ -291,75 +300,84 @@ def _OnlyWithArch(*a, **kw):
     @functools.wraps(function)
-    def _OnlyWithKernelDebugSyms(*a, **kw):
+    def _OnlyWithKernelDebugSyms(*a: Any, **kw: Any) -> Optional[T]:
         if pwndbg.gdblib.kernel.has_debug_syms():
             return function(*a, **kw)
         else:
             print(
                 f"{function.__name__}: This command may only be run when debugging a Linux kernel with debug symbols."
             )
+            return None
 
     return _OnlyWithKernelDebugSyms
 
 
-def OnlyWhenPagingEnabled(function):
+def OnlyWhenPagingEnabled(function: Callable[..., T]) -> Callable[..., Optional[T]]:
     @functools.wraps(function)
-    def _OnlyWhenPagingEnabled(*a, **kw):
+    def _OnlyWhenPagingEnabled(*a: Any, **kw: Any) -> Optional[T]:
         if pwndbg.gdblib.kernel.paging_enabled():
             return function(*a, **kw)
         else:
             print(f"{function.__name__}: This command may only be run when paging is enabled.")
+            return None
 
     return _OnlyWhenPagingEnabled
 
 
-def OnlyWhenRunning(function):
+def OnlyWhenRunning(function: Callable[..., T]) -> Callable[..., Optional[T]]:
     @functools.wraps(function)
-    def _OnlyWhenRunning(*a, **kw):
+    def _OnlyWhenRunning(*a: Any, **kw: Any) -> Optional[T]:
         if pwndbg.gdblib.proc.alive:
             return function(*a, **kw)
         else:
             print(f"{function.__name__}: The program is not being run.")
+            return None
 
     return _OnlyWhenRunning
 
 
-def OnlyWithTcache(function):
+def OnlyWithTcache(function: Callable[..., T]) -> Callable[..., Optional[T]]:
     @functools.wraps(function)
-    def _OnlyWithTcache(*a, **kw):
+    def _OnlyWithTcache(*a: Any, **kw: Any) -> Optional[T]:
         if pwndbg.heap.current.has_tcache():
             return function(*a, **kw)
         else:
             print(
                 f"{function.__name__}: This version of GLIBC was not compiled with tcache support."
             )
+            return None
 
     return _OnlyWithTcache
 
 
-def OnlyWhenHeapIsInitialized(function):
+def OnlyWhenHeapIsInitialized(function: Callable[..., T]) -> Callable[..., Optional[T]]:
     @functools.wraps(function)
-    def _OnlyWhenHeapIsInitialized(*a, **kw):
+    def _OnlyWhenHeapIsInitialized(*a: Any, **kw: Any) -> Optional[T]:
         if pwndbg.heap.current.is_initialized():
             return function(*a, **kw)
         else:
             print(f"{function.__name__}: Heap is not initialized yet.")
+            return None
 
     return _OnlyWhenHeapIsInitialized
 
 
 # TODO/FIXME: Move this elsewhere? Have better logic for that? Maybe caching?
 def _is_statically_linked() -> bool:
-    out = gdb.execute("info dll", to_string=True)
-    return "No shared libraries loaded at this time." in out
+    is_static = False
+    dll_info = gdb.execute("info dll", to_string=True)
+    if dll_info is not None:
+        is_static = "No shared libraries loaded at this time." in dll_info
+    return is_static
```

Comment:
any reason we change this logic here? i think gdb.execute(info dll) should never return None

Reply:
The type hint of gdb.execute is str or None, that's the reason.

Reply:
fixed!